lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -382,6 +431,32 @@ WasmBinaryReader::CallNode()
     m_currentNode.var.num = funcNum;
 }
 
+void
+WasmBinaryReader::CallImportNode()
+{
+    UINT length = 0;
+    UINT32 funcNum = LEB128(length);
+    m_funcState.count += length;
+    if (funcNum >= m_moduleInfo->GetImportCount())
```

Comment:
Now that you mention it I'm not sure why check was here to begin with. I think in WasmBytecodeGenerator we already have this (or we should). Generally, I think semantic validation should not be done in the binary decoder. I'll factor this back out.